File: src/main/scala/com/shockang/study/algorithm/scala/archive/sort/Sorter.scala
package com.shockang.study.algorithm.scala.archive.sort

trait Sorter {
  def sort(a: Array[Int])

  protected def swap(a: Array[Int], i: Int, j: Int): Unit = {
    val tmp = a(i)
    a(i) = a(j)
    a(j) = tmp
  }
}

File: src/main/java/com/shockang/study/algorithm/java/archive/sort/Sorter.java
package com.shockang.study.algorithm.java.archive.sort;

/**
 * @author Shockang
 */
public interface Sorter {
	void sort(int[] a);

	default void swap(int[] a, int i, int j) {
		int tmp = a[i];
		a[i] = a[j];
		a[j] = tmp;
	}
}
